### PR TITLE
feat(treesitter)!: remove silent option from language.add()

### PR DESCRIPTION
**Pull request by lewis6991**
Simply use `pcall` if you want to silence an error.

### CODE REVIEW

**Review by justinmk — APPROVED** [review]
Absolutely agreed.

In many cases, also consider returning `nil` on error, then the caller can opt-into "error" by doing `assert(foo())`.

**Comment by clason** [comment]
Hmm, the current "sanitization" kinda relies on that (see test failures) to avoid spamming the buffer while editing.

**Comment by lewis6991** [comment]
Note: `require_language` remains unchanged. The test failures were because we were passing `silent = true` into `add()`.

File: runtime/lua/vim/treesitter/language.lua [hunk 1]
```diff
@@ -64,32 +68,18 @@ function M.add(lang, opts)
...
 
-  if silent then
-    if not pcall(vim._ts_add_language, path, lang, symbol_name) then
-      return false
-    end
-  else
-    vim._ts_add_language(path, lang, symbol_name)
-  end
-
-  return true
```
[the remark this answers is not in the document]
Comment (clason) [reply runtime/lua/vim/treesitter/language.lua:92]: Otherwise `treesitter/health.lua` has to be adapted.

[the remark this answers is not in the document]
Comment (lewis6991) [reply runtime/lua/vim/treesitter/language.lua:92]: There is an error in `treesitter/health.lua`.

```lua
    local is_loadable, ret = pcall(ts.language.add, parsername)

    if not is_loadable or not ret then
      health.report_error(
        string.format('Parser "%s" failed to load (path: %s): %s', parsername, parser, ret or '?')
      )
    elseif ret then
      local lang = ts.language.inspect_language(parsername)
      health.report_ok(
        string.format('Parser: %-10s ABI: %d, path: %s', parsername, lang._abi_version, parser)
      )
    end
```

If `is_loadable` is `true`, then `ret` will be `nil`. Just need to fix this.

[the remark this answers is not in the document]
Comment (lewis6991) [reply runtime/lua/vim/treesitter/language.lua:92]: #22431

**Pull request merged**